### PR TITLE
Port to Windows

### PR DESCRIPTION
**Pull request by StephenDiligent**
This change set addresses https://github.com/simplesurance/baur/issues/94 and ports baur to Windows. The `make release` command will create a _baur.exe_ file.
Things to note:
1. A [breaking change](https://github.com/golang/go/issues/34610) was introduced to golang.org/x/sys that prevents github.com/docker/docker from compiling. This issue in github.com/docker/docker will not be [fixed](https://github.com/moby/moby/pull/40021) until v20 of github.com/docker/docker is released. In the meantime I have pinned golang.org/x/sys to the latest version before the breaking change was introduced.
2. Some of the existing tests used commands specific to Unix. These have either been replaced with equivalent commands available directly on both Unix and Windows or run via `cmd /C` on Windows.
3. Windows does not allow files that are in use to be deleted in the same way as Unix. In some tests files and folders had to be released so the test cleanup succeeds.
4. Expected paths have been updated to use OS specific path separators where required.
5. CircleCI does not provide Windows containers but does provide Windows VMs
    * The pre-installed version of golang is 1.12.7, this needs to be updated on every build. See [software-pre-installed-in-the-windows-image](https://circleci.com/docs/2.0/hello-world-windows/#software-pre-installed-in-the-windows-image)
    * I could not access Unix docker containers from within the Windows job so postgres is installed on the VM via chocolatey and the default "postgres" database is used.
    * The tests are much slower to run on Windows than Unix, the test timeout has been increased to accommodate this.
    * The `-race` flag has been removed from the Windows `go test` command as a gcc compiler is not installed. I figured this wasn't a problem as it is used in the existing test job. This could be addressed later if necessary.

### CODE REVIEW

**Comment by StephenDiligent** [comment]
@fho disabling the builds on my fork and opening a new PR worked. The checks have now run.
We can also assign this PR to https://github.com/simplesurance/baur/issues/94

**Review by fho — APPROVED** [review]
looks great, thanks a lot for the PR

File: internal/exec/exec_test.go [hunk 1]
```diff
@@ -1,13 +1,30 @@
...
+	"fmt"
+	"runtime"
 	"testing"
 )
 
 func TestEchoStdout(t *testing.T) {
 	const echoStr = "hello world!"
 
-	res, err := Command("echo", "-n", echoStr).Run()
+	// Windows returns StrOutput with surrounding quotation marks
```
[the remark this answers is not in the document]
Comment (StephenDiligent) [reply internal/exec/exec_test.go:12]: I have investigated this further and the results are quite interesting.
I created a [simple app](https://github.com/StephenDiligent/gocommandoutput) that runs a few different commands with a couple of different inputs, I then ran this on Windows and Linux.
Here are the results:
Running on [Windows](https://github.com/StephenDiligent/gocommandoutput/blob/windows/main.go)
image
Running on [Linux](https://github.com/StephenDiligent/gocommandoutput/blob/linux/main.go)
image

Summary:
* **cmd** adds quotation marks around the string if it contains spaces
* **PowerShell** wraps the output onto a new line at the space if it contains spaces
* **more** on Linux outputs headers and footers, Windows doesn't

**Comment by StephenDiligent** [comment]
I noticed that the test results were not being reported in CircleCI. After a bit of investigation I found that the _go-test-report.xml_ file was being generated with Unicode encoding which was causing the problem. 
c6bba2f converts the file to UTF-8 encoding and now the test results are being reported correctly.

**Before:**
image

**After:**
image

File: internal/exec/exec_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (fho) [reply internal/exec/exec_test.go:12]: Ok, so  the `echo` command is probably  also a shell builtin command like for bash and the implementation of it in cmd and PowerShell differ.
When commands are invoked without a shell, the output that is returned by the `exec` package should not be quoted.

[the remark this answers is not in the document]
Comment (StephenDiligent) [reply internal/exec/exec_test.go:12]: That is my understanding. 
Windows doesn't have an _echo_ executable so it must be invoked from a shell. However it does have a _more_ executable and when it is invoked the file contents are not quoted.

**Pull request merged**